### ISSUE TITLE
[Tuning] Standard.ImpossibleTravel - Inconsistent satellite network check in caching logic

### ISSUE THREAD

**Issue by zestysoft** [panther-labs/panther-analysis#1873]
## Rule Information
- **Rule ID**: Standard.ImpossibleTravel
- **File**: `rules/standard_rules/impossible_travel_login.py`

## Issue Description

There is an inconsistency in the caching logic between two locations in the code:

**Line 131** (when there's no prior login history):
```python
if not any((IS_VPN, IS_PRIVATE_RELAY, IS_SATELLITE_NETWORK)):
    put_string_set(...)
```
This correctly excludes VPN, Private Relay, AND Satellite Network logins from being cached.

**Line 159** (when there IS prior login history):
```python
if not IS_PRIVATE_RELAY and not IS_VPN:
    put_string_set(...)
```
This only checks for Private Relay and VPN, but **misses** `IS_SATELLITE_NETWORK`.

## Impact

If a user logs in from a satellite network (e.g., airplane WiFi like GoGo) and there IS existing login history:
1. The satellite network login will be cached (incorrectly)
2. The next ground-based login will be compared against the satellite IP
3. Satellite IPs can geolocate far from actual location → potential false positives

## Suggested Fix

Change line 159 from:
```python
if not IS_PRIVATE_RELAY and not IS_VPN:
```

To (consistent with line 131):
```python
if not any((IS_VPN, IS_PRIVATE_RELAY, IS_SATELLITE_NETWORK)):
```
